code/modules/reagents/chemistry/recipes.dm

```diff
@@ -132,3 +282,107 @@
+	var/datum/reagents/invert_reagents = new (2100, NO_REACT)//I think the biggest size we can get is 2100?
+	var/datum/effect_system/smoke_spread/chem/smoke = new()
+	var/sum_volume = 0
+	invert_reagents.my_atom = holder.my_atom //Give the gas a fingerprint
+	for(var/datum/reagent/reagent in holder.reagent_list) //make gas for reagents, has to be done this way, otherwise it never stops Exploding
+		if(!(reagent.type in required_reagents) || !(reagent.type in results))
+			continue
+		if(reagent.inverse_chem)
+			invert_reagents.add_reagent(reagent.inverse_chem, reagent.volume, no_react = TRUE)
+			holder.remove_reagent(reagent.type, reagent.volume)
+			continue
+		invert_reagents.add_reagent(reagent.type, reagent.volume, added_purity = reagent.purity, no_react = TRUE)
+		sum_volume += reagent.volume
+		holder.remove_reagent(reagent.type, reagent.volume)
+	if(invert_reagents.reagent_list)
+		smoke.set_up(invert_reagents, (sum_volume/5), holder.my_atom)
+		smoke.start()
+	holder.my_atom.audible_message("The [holder.my_atom] suddenly explodes, launching the aerosolized reagents into the air!")
+	if(clear_reactants)
+		clear_reactants(holder)
+	if(clear_products)	
+		clear_products(holder)
+
+//Spews out the corrisponding reactions reagents  (products/required) of the beaker in a smokecloud. Doesn't spew catalysts
+/datum/chemical_reaction/proc/explode_smoke(datum/reagents/holder, datum/equilibrium/equilibrium, clear_products = TRUE, clear_reactants = TRUE)
+	var/datum/reagents/reagents = new/datum/reagents(2100, NO_REACT)//Lets be safe first
+	var/datum/effect_system/smoke_spread/chem/smoke = new()
+	reagents.my_atom = holder.my_atom //fingerprint
+	var/sum_volume = 0
+	for (var/datum/reagent/reagent in holder.reagent_list) 
+		if((reagent.type in required_reagents) || (reagent.type in results))
+			reagents.add_reagent(reagent.type, reagent.volume, added_purity = reagent.purity, no_react = TRUE)
+			holder.remove_reagent(reagent.type, reagent.volume)
+	if(reagents.reagent_list)
+		smoke.set_up(reagents, (sum_volume/5), holder.my_atom)
+		smoke.start()
+	holder.my_atom.audible_message("The [holder.my_atom] suddenly explodes, launching the aerosolized reagents into the air!")
+	if(clear_reactants)
+		clear_reactants(holder)
+	if(clear_products)	
+		clear_products(holder)
+
+//Pushes everything out, and damages mobs with 10 brute damage.
+/datum/chemical_reaction/proc/explode_shockwave(datum/reagents/holder, datum/equilibrium/equilibrium)
+	var/turf/this_turf = get_turf(holder.my_atom)
+	holder.my_atom.audible_message("The [holder.my_atom] suddenly explodes, sending a shockwave rippling through the air!")
+	playsound(this_turf, 'sound/chemistry/shockwave_explosion.ogg', 80, TRUE)
+	//Modified goonvortex
+	for(var/atom/movable/movey in orange(3, this_turf))
+		if(isliving(movey))
+			var/mob/living/live = movey
+			live.adjustBruteLoss(5)//Since this can be called multiple times
```

Comment:
This should respect armor. Use `apply_damage()`.

Reply:
Done!